File: scratch/check_links.py
import os
import re
from bs4 import BeautifulSoup

def check_links():
    html_files = [f for f in os.listdir('.') if f.endswith('.html')]
    print(f"Found {len(html_files)} HTML files.")
    
    # We want to check:
    # 1. Links inside each file.
    # 2. Can we reach every page from index.html (directly or indirectly)?
    # 3. Can we go back to index.html from every page?
    
    file_links = {}
    for filename in html_files:
        filepath = os.path.join('.', filename)
        with open(filepath, 'r', encoding='utf-8', errors='ignore') as f:
            content = f.read()
        
        soup = BeautifulSoup(content, 'html.parser')
        links = []
        for a in soup.find_all('a'):
            href = a.get('href')
            if href:
                # Resolve petite-vue bindings if any, e.g. :href="homeHref"
                # But actually, let's check both regular href and bound :href
                links.append(href)
        
        for a in soup.find_all(attrs=lambda x: x and (x == ':href' or x == 'v-bind:href')):
            # get the value
            val = a.get(':href') or a.get('v-bind:href')
            if val:
                links.append(f"vue:{val}")
                
        file_links[filename] = set(links)
        
    print("\n--- Links in each page ---")
    for file, links in sorted(file_links.items()):
        print(f"\n{file}:")
        # Resolve vue variable bindings if we can deduce them
        resolved = []
        for l in sorted(links):
            if l.startswith("vue:"):
                # E.g. vue:homeHref, vue:aboutHref
                var_name = l.split(":")[1]
                # Look for v-scope in the file to resolve it
                resolved.append(f"{l} (dynamic)")
            else:
                resolved.append(l)
        print("  " + ", ".join(resolved))

if __name__ == '__main__':
    check_links()

File: scratch/check_links.js
const fs = require('fs');
const path = require('path');

function checkLinks() {
    const files = fs.readdirSync('.').filter(f => f.endsWith('.html'));
    console.log(`Found ${files.length} HTML files.`);

    const fileLinks = {};
    const vueVars = {
        homeHref: 'index.html',
        aboutHref: 'about.html',
        eventsHref: 'events.html',
        tournamentsHref: 'tournaments.html',
        membershipHref: 'membership.html',
        coachesHref: 'coaches.html',
        blogHref: 'blog.html',
        contactHref: 'contact.html',
        ctaHref: 'free-trial.html'
    };

    files.forEach(filename => {
        const content = fs.readFileSync(filename, 'utf-8');
        const hrefRegex = /href="([^"]*)"|:href="([^"]*)"/g;
        let match;
        const links = new Set();

        while ((match = hrefRegex.exec(content)) !== null) {
            const rawHref = match[1];
            const vueHref = match[2];

            if (rawHref) {
                links.add(rawHref);
            } else if (vueHref) {
                const resolved = vueVars[vueHref] || `vue:${vueHref}`;
                links.add(resolved);
            }
        }
        fileLinks[filename] = Array.from(links);
    });

    console.log("\n--- Links in each page ---");
    Object.keys(fileLinks).sort().forEach(file => {
        console.log(`\n${file}:`);
        console.log("  " + fileLinks[file].sort().join(", "));
    });
}

checkLinks();
